### PR TITLE
add: Support for downloading artefacts and directories

### PR DESCRIPTION
**Pull request by amritanshu-pandey**

### CODE REVIEW

File: src/nexuscli/cli.py [hunk 1]
```diff
@@ -31,11 +33,13 @@
   -h --help             This screen
   --blob=<store_name>   Use this blob with new repository  [default: default]
   --depth=<repo_depth>  Depth (0-5) where repodata folder(s) exist [default: 0]
+  --flatten             Flatten directory structure on `rekt ar` transfers.
   --force, -f           Execute action without confirmation
   --write=<w_policy>    Accepted: allow, allow_once, deny [default: allow_once]
   --layout=<l_policy>   Accepted: strict, permissive [default: strict]
   --version=<v_policy>  Accepted: release, snapshot, mixed [default: release]
   --strict-content      Enable strict content type validation
+  --log_lvl=<log_lvl>   Logging level [default: WARNING].
```
Comment (thiagofigueiro) [inline src/nexuscli/cli.py]: Missing entry for `--nocache`. Also please try to keep in alphabetical order.

File: src/nexuscli/cli.py [hunk 2]
```diff
@@ -238,6 +242,31 @@ def cmd_upload(args):
...
+def cmd_download(args):
+    """Performs ``nexus3 download``"""
+    nexus_client = get_client()
+    source = args['<from_repository>']
+    destination = args['<to_dst>']
+
+    sys.stderr.write(
+        'Downloading {source} to {destination}\n'.format(**locals()))
+
+    FLATTEN = args.get('--flatten')
```
Comment (thiagofigueiro) [inline src/nexuscli/cli.py]: UPPERCASE are used for constants. Kind of weird to see these use-once variables here.

How about doing the `args.get()` directly as a kw argument of `download()`?  Break some lines - that's fine.

File: src/nexuscli/nexus_client.py [hunk 3]
```diff
@@ -483,3 +498,163 @@ def upload(self, source, destination):
...
+            service.
+        :param destination: path to the local file or directory.
+        :param flatten: when True, the remote path isn't reproduced locally.
+        :param nocache: Force download of a directory or artefact even if local
+                        copy is available and is up-to-date with the version
+                        available on Nexus.
+        :return: number of downloaded files.
+        """
+        # FIXME: Enable logging at application level rather than in this method
+        LOG.setLevel(level=log_lvl)
```
Comment (thiagofigueiro) [inline src/nexuscli/nexus_client.py]: This presentation from PyCon au 2018 (by @funkybob) might be useful. The "ah-ha" moment for me was that log levels can be set both at the logger and handler objects and the lowest wins (ie. if you set one to debug and the other to warning, you'll only see warning and above).

https://www.youtube.com/watch?v=DxZ5WEo4hvU&feature=youtu.be&t=4m20s

File: src/nexuscli/nexus_client.py [hunk 4]
```diff
@@ -483,3 +498,163 @@ def upload(self, source, destination):
...
+                'Downloading from {download_url}. '
+                'Reason: {response.reason}'.format(**locals()))
+
+        with open(destination, 'wb') as fd:
+            LOG.debug('Writing {download_url} to {destination}\n'.format(
+                **locals()))
+            for chunk in response.iter_content():
+                fd.write(chunk)
+
+    def download(self, source, destination, log_lvl='WARNING',
```
Comment (thiagofigueiro) [inline src/nexuscli/nexus_client.py]: Could you please create an issue in github to refactor this method signature? `log_lvl` doesn't belong here.

Comment (amritanshu-pandey) [reply src/nexuscli/nexus_client.py]: This should not be required as now I have configured logging at application level and have refactored this method.

**Pull request merged**